### DISCUSSION TITLE
飞浆自定义训练识别数据集加载报错

### DISCUSSION THREAD

**Discussion by Eason-shu** [PaddlePaddle/PaddleOCR#13605] [Q&A]
划分数据集的时候，但是载检测模型训练的时候正常
Initialize indexs of datasets:['./train_data/rec/train.txt'] list index out
我看网上是说CPU版本的问题，但是我改成GPU一样的，这个会影响训练效果吗？

**Comment by GreatV** [comment]
配置文件，环境，训练指令等分别是啥样的？

**Reply by Eason-shu** [reply]
-- 环境
Python Version:
Python 3.12.4
PaddlePaddle Version:
3.0.0-beta1
-- 数据集划分指令
`bash`
@echo off
setlocal enabledelayedexpansion

:: 打印系统信息
echo -------- System Information --------
echo Current Date and Time: !date! !time!
echo Python Version:
python --version

:: 指定数据分割脚本的路径和参数
set SCRIPT=gen_ocr_train_val_test.py
set TRAINVALTESTRATIO=6:2:2
set DATASETROOTPATH=E:\PPOCRLabel-main\train_data
set DETROOTPATH=E:\PPOCRLabel-main\train_data\det
set RECROOTPATH=E:\PPOCRLabel-main\train_data\rec

:: 打印数据集分割命令
echo --------
echo Data Split Command:
echo python !SCRIPT! --trainValTestRatio !TRAINVALTESTRATIO! --datasetRootPath !DATASETROOTPATH! --detRootPath !DETROOTPATH! --recRootPath !RECROOTPATH!

:: 执行Python数据集分割命令
echo --------
echo -------- begin data split --------
python !SCRIPT! --trainValTestRatio !TRAINVALTESTRATIO! --datasetRootPath !DATASETROOTPATH! --detRootPath !DETROOTPATH! --recRootPath !RECROOTPATH!
echo -------- end data split --------
endlocal
-- 配置文件 ch_PP-OCRv3_rec_distillation.yml
Global:
  debug: true # debug model
  use_gpu: false
  epoch_num: 800
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/rec_ppocr_v3_distillation
  save_epoch_step: 100
  eval_batch_step: [0, 10]
  cal_metric_during_train: true
  pretrained_model: ./pretrain_models/ch_PP-OCRv3_rec_train/ch_PP-OCRv3_rec_train/best_accuracy #设置加载预训练模型路径
  checkpoints:
  save_inference_dir:
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  max_text_length: &max_text_length 25
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ppocrv3_distillation.txt
  d2s_train_image_shape: [3, 48, -1]


Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Piecewise
    decay_epochs : [700]
    values : [0.0005, 0.00005]
    warmup_epoch: 5
  regularizer:
    name: L2
    factor: 3.0e-05


Architecture:
  model_type: &model_type "rec"
  name: DistillationModel
  algorithm: Distillation
  Models:
    Teacher:
      pretrained:
      freeze_params: false
      return_all_feats: true
      model_type: *model_type
      algorithm: SVTR_LCNet
      Transform:
      Backbone:
        name: MobileNetV1Enhance
        scale: 0.5
        last_conv_stride: [1, 2]
        last_pool_type: avg
        last_pool_kernel_size: [2, 2]
      Head:
        name: MultiHead
        head_list:
          - CTCHead:
              Neck:
                name: svtr
                dims: 64
                depth: 2
                hidden_dims: 120
                use_guide: True
              Head:
                fc_decay: 0.00001
          - SARHead:
              enc_dim: 512
              max_text_length: *max_text_length
    Student:
      pretrained:
      freeze_params: false
      return_all_feats: true
      model_type: *model_type
      algorithm: SVTR_LCNet
      Transform:
      Backbone:
        name: MobileNetV1Enhance
        scale: 0.5
        last_conv_stride: [1, 2]
        last_pool_type: avg
        last_pool_kernel_size: [2, 2]
      Head:
        name: MultiHead
        head_list:
          - CTCHead:
              Neck:
                name: svtr
                dims: 64
                depth: 2
                hidden_dims: 120
                use_guide: True
              Head:
                fc_decay: 0.00001
          - SARHead:
              enc_dim: 512
              max_text_length: *max_text_length
Loss:
  name: CombinedLoss
  loss_config_list:
    - DistillationDMLLoss:
        weight: 1.0
        act: "softmax"
        use_log: true
        model_name_pairs:
          - ["Student", "Teacher"]
        key: head_out
        multi_head: True
        dis_head: ctc
        name: dml_ctc
    - DistillationDMLLoss:
        weight: 0.5
        act: "softmax"
        use_log: true
        model_name_pairs:
          - ["Student", "Teacher"]
        key: head_out
        multi_head: True
        dis_head: sar
        name: dml_sar
    - DistillationDistanceLoss:
        weight: 1.0
        mode: "l2"
        model_name_pairs:
          - ["Student", "Teacher"]
        key: backbone_out
    - DistillationCTCLoss:
        weight: 1.0
        model_name_list: ["Student", "Teacher"]
        key: head_out
        multi_head: True
    - DistillationSARLoss:
        weight: 1.0
        model_name_list: ["Student", "Teacher"]
        key: head_out
        multi_head: True

PostProcess:
  name: DistillationCTCLabelDecode
  model_name: ["Student", "Teacher"]
  key: head_out
  multi_head: True

Metric:
  name: DistillationMetric
  base_metric_name: RecMetric
  main_indicator: acc
  key: "Student"
  ignore_space: False

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/
    ext_op_transform_idx: 1
    label_file_list:
      - ./train_data/rec/train.txt # 训练数据标签文件
    transforms:
      - DecodeImage:
          img_mode: BGR
          channel_first: false
      - RecConAug:
          prob: 0.5
          ext_data_num: 2
          image_shape: [48, 320, 3]
          max_text_length: *max_text_length
      - RecAug:
      - MultiLabelEncode:
      - RecResizeImg:
          image_shape: [3, 48, 320]
      - KeepKeys:
          keep_keys:
            - image
            - label_ctc
            - label_sar
            - length
            - valid_ratio
  loader:
    shuffle: true
    batch_size_per_card: 128
    drop_last: true
    num_workers: 4
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data
    label_file_list:
      - ./train_data/rec/val.txt # 训练数据标签文件
    transforms:
      - DecodeImage:
          img_mode: BGR
          channel_first: false
      - MultiLabelEncode:
      - RecResizeImg:
          image_shape: [3, 48, 320]
      - KeepKeys:
          keep_keys:
            - image
            - label_ctc
            - label_sar
            - length
            - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 128
    num_workers: 4
-- 训练指令
@echo off
setlocal enabledelayedexpansion

:: 指定Python解释器的路径，如果不在PATH中
:: set PYTHON_EXECUTABLE=C:\path\to\python.exe

:: 打印系统信息
echo -------- System Information --------
echo Current Date and Time: !date! !time!
echo Python Version:
python --version
echo PaddlePaddle Version:
python -c "import paddle; print(paddle.__version__)"

:: 指定训练脚本和配置文件的路径
set TRAIN_SCRIPT=tools\train.py
set CONFIG_FILE=pretrain_models\ch_PP-OCRv3_rec_distillation.yml

:: 打印训练命令
echo --------
echo Training Command:
echo python !TRAIN_SCRIPT! -c !CONFIG_FILE!

:: 执行Python训练命令
echo --------
echo -------- begin train --------
python !TRAIN_SCRIPT! -c !CONFIG_FILE!
echo -------- end train --------

endlocal

**Reply by GreatV** [reply]
参考一下这里 https://github.com/PaddlePaddle/PaddleOCR/issues/13527

**Reply by GreatV** [reply]
是不是数据量太少了呀

**Reply by Eason-shu** [reply]
好的，感谢解答，我试试
